### PR TITLE
Add `.phar` extension to the binary

### LINKED ISSUE #6711: Add `.phar` to the filename in the shim release
## Feature request

It would be nice if the [shim](https://github.com/PHP-CS-Fixer/shim) file was called `php-cs-fixer.phar` instead of `php-cs-fixer` so PhpStorm parses the context of the phar file and does not complain when working on `.php-cs-fixer.dist.php` that `PhpCsFixer\Finder` or `PhpCsFixer\Config` does not exist. Also auto-complete for those classes would then work as well.

If you want to keep the `vendor/bin/php-cs-fixer` file maybe you could do the same thins as phpstan does, see [here](https://github.com/phpstan/phpstan/blob/bf4d335fbbba4dbb98dac4bd82fcf0fa03907e7d/composer.json#L12-L13) and [here](https://github.com/phpstan/phpstan/blob/bf4d335fbbba4dbb98dac4bd82fcf0fa03907e7d/phpstan).

### PR DESCRIPTION
**Pull request by gjuric**
Resolves https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/6711

### CODE REVIEW

**Review by localheinz — APPROVED** [review]

**Comment by localheinz** [comment]
Thank you, @gjuric!

**Comment by keradus** [comment]
@gjuric @localheinz , as we now have `php-cs-fixer` and `php-cs-fixer.phar`, should we rename `php-cs-fixer.asc` to `php-cs-fixer.phar.asc`, to show which file got signed?

**Comment by localheinz** [comment]
@keradus 

Makes sense to me!

**Comment by localheinz** [comment]
@keradus 

See https://github.com/PHP-CS-Fixer/shim/pull/3!

File: composer.json [hunk 1]
```diff
@@ -26,7 +26,8 @@
         "ext-mbstring": "For handling non-UTF8 characters."
     },
     "bin": [
-        "php-cs-fixer"
+        "php-cs-fixer",
+        "php-cs-fixer.phar",
```
Comment (keradus) [inline composer.json:30]: Screenshot 2023-01-09 at 13 01 11

Comment (keradus) [reply composer.json:30]: https://github.com/PHP-CS-Fixer/shim/pull/6

**Pull request merged**